### PR TITLE
Follow artist

### PR DESCRIPTION
**Pull request by ashleyjelks**
This PR adds the support to and handle mutations to follow/unfollow artists in the Collections featured artist component.

Kapture 2020-01-23 at 12 48 09

https://artsyproduct.atlassian.net/browse/FX-1659

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>
<b>🃏 FAIL</b> in <a href="https://github.com/artsy/emission/blob/follow-artist/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx">FeaturedArtists-tests.tsx</a>


<li>
<a href='https://github.com/artsy/emission/blob/follow-artist/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx#L86'>renders properly</a>
<br/>
Snapshot name: `FeaturedArtists renders properly 1`- Snapshot+ Received- "<view pb="1"><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" pb="15">Featured Artists</text><view flexwrap="wrap"><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Pablo Picasso</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">American, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Andy Warhol</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">American, b. 1947</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="secondaryOutline" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#000">Follow</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Joan Miro</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">Spanish, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view></view></view>"+ "<view pb="1"><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" pb="15">Featured Artists</text><view flexwrap="wrap"><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Joan Miro</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">Spanish, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Joan Miro</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">Spanish, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Joan Miro</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">Spanish, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view></view></view>"

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(received).toMatchSnapshot()

Snapshot name: `FeaturedArtists renders properly 1`

- Snapshot
+ Received

- "<view pb="1"><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" pb="15">Featured Artists</text><view flexwrap="wrap"><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Pablo Picasso</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">American, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Andy Warhol</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">American, b. 1947</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="secondaryOutline" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#000">Follow</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Joan Miro</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">Spanish, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view></view></view>"
+ "<view pb="1"><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" pb="15">Featured Artists</text><view flexwrap="wrap"><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Joan Miro</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">Spanish, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Joan Miro</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">Spanish, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view><view width="100%" pb="20"><view flexdirection="row" justifycontent="space-between" flexwrap="nowrap" focusable="true"><view mr="1" justifycontent="center"><image resizemode="stretch" style="width: 45px; height: 45px; border-radius: 22.5px;" source="[object Object]"></image></view><view justifycontent="center" width="0" flexgrow="1"><text font-family="ReactNativeAGaramondPro-Regular" font-size="16px" lineheight="20px" ellipsizemode="tail" numberoflines="1" color="black100">Joan Miro</text><text font-family="Unica77LL-Regular" font-size="12px" lineheight="16px" ellipsizemode="tail" numberoflines="1" color="black60">Spanish, b. 1877</text></view><view ml="1" flexdirection="row" alignitems="center" justifycontent="flex-end"><view flexdirection="row" focusable="true"><view variant="primaryBlack" theme="[object Object]" px="1"><view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" color="#FFF">Following</text></view><text font-family="Unica77LL-Medium" font-size="12px" lineheight="16px" role="presentation">Following</text></view></view></view></view></view></view></view>"
    at toMatchSnapshot (/home/travis/build/artsy/emission/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx:86:25)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.<computed> [as next] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:97:21)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at invoke (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:135:20)
    at /home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:145:13
    at /home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/modules/es.promise.js:113:22
    at flush (/home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/internals/microtask.js:26:9)
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
</code></pre>
</details>
</li>
<br/>



<li>
<a href='https://github.com/artsy/emission/blob/follow-artist/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx#L96'>renders an EntityHeader for each featured artist</a>
<br/>
Expected substring: "Pablo Picasso"Received string: "<view pb=\"1\"><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" pb=\"15\">Featured Artists</text><view flexwrap=\"wrap\"><view width=\"100%\" pb=\"20\"><view flexdirection=\"row\" justifycontent=\"space-between\" flexwrap=\"nowrap\" focusable=\"true\"><view mr=\"1\" justifycontent=\"center\"><image resizemode=\"stretch\" style=\"width: 45px; height: 45px; border-radius: 22.5px;\" source=\"[object Object]\"></image></view><view justifycontent=\"center\" width=\"0\" flexgrow=\"1\"><text font-family=\"ReactNativeAGaramondPro-Regular\" font-size=\"16px\" lineheight=\"20px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black100\">Joan Miro</text><text font-family=\"Unica77LL-Regular\" font-size=\"12px\" lineheight=\"16px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black60\">Spanish, b. 1877</text></view><view ml=\"1\" flexdirection=\"row\" alignitems=\"center\" justifycontent=\"flex-end\"><view flexdirection=\"row\" focusable=\"true\"><view variant=\"primaryBlack\" theme=\"[object Object]\" px=\"1\"><view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" color=\"#FFF\">Following</text></view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" role=\"presentation\">Following</text></view></view></view></view></view><view width=\"100%\" pb=\"20\"><view flexdirection=\"row\" justifycontent=\"space-between\" flexwrap=\"nowrap\" focusable=\"true\"><view mr=\"1\" justifycontent=\"center\"><image resizemode=\"stretch\" style=\"width: 45px; height: 45px; border-radius: 22.5px;\" source=\"[object Object]\"></image></view><view justifycontent=\"center\" width=\"0\" flexgrow=\"1\"><text font-family=\"ReactNativeAGaramondPro-Regular\" font-size=\"16px\" lineheight=\"20px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black100\">Joan Miro</text><text font-family=\"Unica77LL-Regular\" font-size=\"12px\" lineheight=\"16px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black60\">Spanish, b. 1877</text></view><view ml=\"1\" flexdirection=\"row\" alignitems=\"center\" justifycontent=\"flex-end\"><view flexdirection=\"row\" focusable=\"true\"><view variant=\"primaryBlack\" theme=\"[object Object]\" px=\"1\"><view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" color=\"#FFF\">Following</text></view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" role=\"presentation\">Following</text></view></view></view></view></view><view width=\"100%\" pb=\"20\"><view flexdirection=\"row\" justifycontent=\"space-between\" flexwrap=\"nowrap\" focusable=\"true\"><view mr=\"1\" justifycontent=\"center\"><image resizemode=\"stretch\" style=\"width: 45px; height: 45px; border-radius: 22.5px;\" source=\"[object Object]\"></image></view><view justifycontent=\"center\" width=\"0\" flexgrow=\"1\"><text font-family=\"ReactNativeAGaramondPro-Regular\" font-size=\"16px\" lineheight=\"20px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black100\">Joan Miro</text><text font-family=\"Unica77LL-Regular\" font-size=\"12px\" lineheight=\"16px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black60\">Spanish, b. 1877</text></view><view ml=\"1\" flexdirection=\"row\" alignitems=\"center\" justifycontent=\"flex-end\"><view flexdirection=\"row\" focusable=\"true\"><view variant=\"primaryBlack\" theme=\"[object Object]\" px=\"1\"><view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" color=\"#FFF\">Following</text></view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" role=\"presentation\">Following</text></view></view></view></view></view></view></view>"

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(received).toContain(expected) // indexOf

Expected substring: "Pablo Picasso"
Received string:    "<view pb=\"1\"><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" pb=\"15\">Featured Artists</text><view flexwrap=\"wrap\"><view width=\"100%\" pb=\"20\"><view flexdirection=\"row\" justifycontent=\"space-between\" flexwrap=\"nowrap\" focusable=\"true\"><view mr=\"1\" justifycontent=\"center\"><image resizemode=\"stretch\" style=\"width: 45px; height: 45px; border-radius: 22.5px;\" source=\"[object Object]\"></image></view><view justifycontent=\"center\" width=\"0\" flexgrow=\"1\"><text font-family=\"ReactNativeAGaramondPro-Regular\" font-size=\"16px\" lineheight=\"20px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black100\">Joan Miro</text><text font-family=\"Unica77LL-Regular\" font-size=\"12px\" lineheight=\"16px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black60\">Spanish, b. 1877</text></view><view ml=\"1\" flexdirection=\"row\" alignitems=\"center\" justifycontent=\"flex-end\"><view flexdirection=\"row\" focusable=\"true\"><view variant=\"primaryBlack\" theme=\"[object Object]\" px=\"1\"><view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" color=\"#FFF\">Following</text></view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" role=\"presentation\">Following</text></view></view></view></view></view><view width=\"100%\" pb=\"20\"><view flexdirection=\"row\" justifycontent=\"space-between\" flexwrap=\"nowrap\" focusable=\"true\"><view mr=\"1\" justifycontent=\"center\"><image resizemode=\"stretch\" style=\"width: 45px; height: 45px; border-radius: 22.5px;\" source=\"[object Object]\"></image></view><view justifycontent=\"center\" width=\"0\" flexgrow=\"1\"><text font-family=\"ReactNativeAGaramondPro-Regular\" font-size=\"16px\" lineheight=\"20px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black100\">Joan Miro</text><text font-family=\"Unica77LL-Regular\" font-size=\"12px\" lineheight=\"16px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black60\">Spanish, b. 1877</text></view><view ml=\"1\" flexdirection=\"row\" alignitems=\"center\" justifycontent=\"flex-end\"><view flexdirection=\"row\" focusable=\"true\"><view variant=\"primaryBlack\" theme=\"[object Object]\" px=\"1\"><view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" color=\"#FFF\">Following</text></view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" role=\"presentation\">Following</text></view></view></view></view></view><view width=\"100%\" pb=\"20\"><view flexdirection=\"row\" justifycontent=\"space-between\" flexwrap=\"nowrap\" focusable=\"true\"><view mr=\"1\" justifycontent=\"center\"><image resizemode=\"stretch\" style=\"width: 45px; height: 45px; border-radius: 22.5px;\" source=\"[object Object]\"></image></view><view justifycontent=\"center\" width=\"0\" flexgrow=\"1\"><text font-family=\"ReactNativeAGaramondPro-Regular\" font-size=\"16px\" lineheight=\"20px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black100\">Joan Miro</text><text font-family=\"Unica77LL-Regular\" font-size=\"12px\" lineheight=\"16px\" ellipsizemode=\"tail\" numberoflines=\"1\" color=\"black60\">Spanish, b. 1877</text></view><view ml=\"1\" flexdirection=\"row\" alignitems=\"center\" justifycontent=\"flex-end\"><view flexdirection=\"row\" focusable=\"true\"><view variant=\"primaryBlack\" theme=\"[object Object]\" px=\"1\"><view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" color=\"#FFF\">Following</text></view><text font-family=\"Unica77LL-Medium\" font-size=\"12px\" lineheight=\"16px\" role=\"presentation\">Following</text></view></view></view></view></view></view></view>"
    at toContain (/home/travis/build/artsy/emission/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx:96:20)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.<computed> [as next] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:97:21)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at invoke (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:135:20)
    at /home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:145:13
    at /home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/modules/es.promise.js:113:22
    at flush (/home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/internals/microtask.js:26:9)
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
</code></pre>
</details>
</li>
<br/>



<li>
<a href='https://github.com/artsy/emission/blob/follow-artist/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx#L108'>does not render an EntityHeader for excluded artists</a>
<br/>
Expected: 1, received: 3

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(received).toEqual(expected) // deep equality

Expected: 1
Received: 3
    at toEqual (/home/travis/build/artsy/emission/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx:108:34)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.<computed> [as next] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:97:21)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at invoke (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:135:20)
    at /home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:145:13
    at /home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/modules/es.promise.js:113:22
    at flush (/home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/internals/microtask.js:26:9)
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
</code></pre>
</details>
</li>
<br/>



<li>
<a href='https://github.com/artsy/emission/blob/follow-artist/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx#L124'>does not render an EntityHeader for any non-requested artists</a>
<br/>
Expected: 1, received: 0

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(received).toEqual(expected) // deep equality

Expected: 1
Received: 0
    at toEqual (/home/travis/build/artsy/emission/src/lib/Scenes/Collection/Components/__tests__/FeaturedArtists-tests.tsx:124:36)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.<computed> [as next] (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:97:21)
    at tryCatch (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:45:40)
    at invoke (/home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:135:20)
    at /home/travis/build/artsy/emission/node_modules/regenerator-runtime/runtime.js:145:13
    at /home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/modules/es.promise.js:113:22
    at flush (/home/travis/build/artsy/emission/node_modules/react-tracking/node_modules/core-js/internals/microtask.js:26:9)
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
</code></pre>
</details>
</li>
<br/>

</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Scenes/Collection/Components/FeaturedArtists.tsx [hunk 1]
```diff
@@ -69,17 +75,78 @@ export class FeaturedArtists extends React.Component<FeaturedArtistsProps, Featu
     return allArtists
   }
 
+  failedFollowChange() {
+    // callback for analytics purposes
```
Comment (anandaroop) [inline src/lib/Scenes/Collection/Components/FeaturedArtists.tsx]: The tracking calls are still to come, correct? (It's ticketed elsewhere, so just double-checking)

File: src/lib/Scenes/Collection/Components/FeaturedArtists.tsx [hunk 2]
```diff
@@ -69,17 +75,78 @@ export class FeaturedArtists extends React.Component<FeaturedArtistsProps, Featu
...
+    this.setState({
+      isFollowedChanging: false,
+    })
+  }
+
+  successfulFollowChange() {
+    // callback for analytics purposes
+    Events.userHadMeaningfulInteraction()
```
Comment (anandaroop) [inline src/lib/Scenes/Collection/Components/FeaturedArtists.tsx]: Out of curiosity, I see this is related to requesting App Store reviews — any  advice on what constitutes a "meaningful interaction"?

File: src/lib/Scenes/Collection/Components/FeaturedArtists.tsx [hunk 3]
```diff
@@ -69,17 +75,78 @@ export class FeaturedArtists extends React.Component<FeaturedArtistsProps, Featu
...
+              }
+            }
+          `,
+          variables: {
+            input: {
+              artistID: slug,
+              unfollow: isFollowed,
+            },
+          },
+          optimisticResponse: {
```
Comment (anandaroop) [inline src/lib/Scenes/Collection/Components/FeaturedArtists.tsx]: Ah, neat. I was going to ask why we don't see a spinner when we toggle the follow buttons in the entity header — I take it this is why?

File: src/lib/relay/QueryRenderers.tsx [hunk 4]
```diff
@@ -284,15 +284,16 @@ export const CollectionRenderer: React.SFC<CollectionRendererProps> = ({ collect
...
-          ...Collection_collection @arguments(screenWidth: $screenWidth)
+          ...Collection_collection @arguments(screenWidth: $screenWidth, headerHeight: $headerHeight)
         }
       }
     `}
     variables={{
       collectionID,
-      screenWidth: Dimensions.get("screen").width,
+      screenWidth: Dimensions.get("screen").width > 500 ? Dimensions.get("screen").width : 500,
+      headerHeight: Dimensions.get("screen").height > 700 ? Dimensions.get("screen").height : 204,
```
Comment (anandaroop) [inline src/lib/relay/QueryRenderers.tsx]: The dimensions stuff is intentionally part of this PR, yeah?

Comment (ashleyjelks) [reply src/lib/relay/QueryRenderers.tsx]: Yes, it was a small enough change to re-size the header in the same PR so I included it.

File: src/lib/Scenes/Collection/Components/FeaturedArtists.tsx (hunk 1 above)
Comment (ashleyjelks) [reply src/lib/Scenes/Collection/Components/FeaturedArtists.tsx]: Yes, we still need to add tracking per what @abhitip outlined in the Jira ticket!

**Comment by pepopowitz** [comment]
Per a slack conversation, @sweir27 pointed out that [there's an existing component](https://github.com/artsy/emission/blob/master/src/lib/Components/ArtistListItem.tsx) we could use instead of [generating each artist item ourselves](https://github.com/artsy/emission/pull/2054/files#diff-cf72bf49360aa2b6be4c6407c4b8ce68L26). I think we should leave this PR open but mark it "do not merge", until we're certain that reusing the ArtistListItem component works. 

Thoughts?

**Comment by ashleyjelks** [comment]
> Per a slack conversation, @sweir27 pointed out that [there's an existing component](https://github.com/artsy/emission/blob/master/src/lib/Components/ArtistListItem.tsx) we could use instead of [generating each artist item ourselves](https://github.com/artsy/emission/pull/2054/files#diff-cf72bf49360aa2b6be4c6407c4b8ce68L26). I think we should leave this PR open but mark it "do not merge", until we're certain that reusing the ArtistListItem component works.
> 
> Thoughts?

Sounds good to me! We'll need to make a new ticket reflect this migration work. Also, there's a small change to resize the header image that needs to be put into a separate PR. I will make that change now as well since we likely won't be merging this one.

**Pull request closed without merging**